pkg/statistics/handle/storage/read.go

```diff
@@ -621,28 +621,25 @@ func loadNeededColumnHistograms(sctx sessionctx.Context, statsHandle statstypes.
 	if !ok {
 		return nil
 	}
+
 	var colInfo *model.ColumnInfo
 	_, loadNeeded, analyzed := tbl.ColumnIsLoadNeeded(col.ID, true)
 	if !loadNeeded || !analyzed {
 		asyncload.AsyncLoadHistogramNeededItems.Delete(col)
 		return nil
 	}
-	isUpdateColAndIdxExistenceMap := false
-	colInfo = tbl.ColAndIdxExistenceMap.GetCol(col.ID)
+
+	// Now, we cannot init the column info in the ColAndIdxExistenceMap when to disable lite-init-stats.
+	// so we have to get the column info from the domain.
+	is := sctx.GetDomainInfoSchema().(infoschema.InfoSchema)
+	tblInfo, ok := statsHandle.TableInfoByID(is, col.TableID)
```

Comment:
What's the difference... it's still calling TableInfoByID?
How could this avoid using infoschema when init stats

Reply:
In this context, stats are loaded only for the tables that require statistical information, thereby avoiding a full load meta.